app/utils/graphql.ts

```diff
@@ -0,0 +1,194 @@
+import { useState, useCallback, useEffect, useMemo } from "react";
+import useSWR, { SWRConfiguration } from "swr";
+import {
+  gql,
+  GraphQLClient,
+  request,
+  RequestDocument,
+  Variables,
+} from "graphql-request";
+import { getHasuraRoleName } from "./auth";
+import { MutationVariables, LookupTableDef } from "@/types/types";
+import { useAuth0 } from "@auth0/auth0-react";
+
+const ENDPOINT = process.env.NEXT_PUBLIC_HASURA_ENDPOINT!;
+
+const DEFAULT_SWR_OPTIONS: SWRConfiguration = {
+  /**
+   * Dont refetch when the page/tab is refocused
+   */
+  revalidateOnFocus: false,
+  /**
+   * Dont refetch on network recon
+   */
+  revalidateOnReconnect: false,
+};
```

Comment:
these are just my opinions to cut back on unnecessary API calls—SWR really loves to make API calls.  i encourage y'all to check out the various settings available 🧪